packages/core/src/interpreter.ts

```diff
@@ -1151,8 +1158,11 @@ export class Interpreter<
     source: Subscribable<T>,
     id: string
   ): SpawnedActorRef<any, T> {
+    let emitted: T | undefined = undefined;
+
     const subscription = source.subscribe(
       (value) => {
+        emitted = value;
         this.send(toSCXMLEvent(value, { origin: id }));
```

Comment:
I don't like how this uses a single value for 2 distinct concepts in case of observables. Currently, observables must emit something that extends ` TEvent` and that gets automatically~ sent to the parent as the event. Persisting that as a snapshot seems weird but I also fail to think of a better design around observables that would handle the conversion to actor interface automatically.

Note that the same happens for callbacks but when it comes to machines and promises both things are separate.

Reply:
For invoked observables, yes, the emitted values are the same as events sent back to the parent, which is why I thought it was fine to have this be the snapshot value.

If we did it a different way, we would have to generate events that look like:

```
{
  type: "???",
  data: {/* emitted data */}
}
```

But what would the `event.type` be? (maybe more of a v5 question)

Reply:
That would have to wrap all the events sent back to the parent (otherwise we'd end up with some magic blackbox algorithm for choosing what should be wrapped and what should not be). And that would in-turn prevent sending events of different types back to the parent from observables and also prevent implementing "epics" (observables fed with events by the machine and responding with events to the machine).

So maybe what you have done here is best? or maybe it should just be undefined? Or maybe we should have to types for observables in the future? After all, subjects (for which this makes sense as what they emit is usually state-like) are somewhat different from other observables (which more often than not emit one-time events without expectation of it being persisted as it's only relevant at that point in time). OTOH the latter type is often built using stateful/aggregating operators in the middle so that state is relevant for them but OYAH we can't rehydrate operators "in the middle" anyway.